apps/swirl-docs/src/components/CodePreview/CodePreview.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  SwirlIconCheckStrong,
+  SwirlIconCopy,
+  SwirlIconExpandLess,
+  SwirlIconExpandMore,
+} from "@getflip/swirl-components-react";
+import { ReactNode, useState } from "react";
+import { CodeSandboxButton } from "./CodeSandboxButton";
+import { CopyToClipboard } from "react-copy-to-clipboard";
+import classNames from "classnames";
+import NoSsr from "../Layout/NoSsr";
+import Highlight, { defaultProps } from "prism-react-renderer";
+import theme from "prism-react-renderer/themes/vsDark";
+import { CodeExample } from "./types";
+import { NpmPackageLink } from "./NpmPackageLink";
+import CodePreviewContext from "./CodePreviewContext";
+import { Request } from "./CodePreviewRequestString";
+
+interface CodePreviewProps {
+  codeExample: CodeExample;
+  children?: ReactNode | ReactNode[];
+}
+
+/**
+ * Let's you easily render syntax highlighted code.
+ * Children of the Component will be displayed in the Header.
+ */
+export function CodePreview({ children, codeExample }: CodePreviewProps) {
+  const [isExpanded, setIsExpanded] = useState<boolean>(false);
+  const [isCopied, setIsCopied] = useState<boolean>(false);
+
+  return (
+    <NoSsr>
+      <CodePreviewContext.Provider value={codeExample}>
```

Comment:
**question:** Is this really necessary? Looking at the usages of `useCodePreviewContext` it seems like you could achieve the same by passing the the code example via props.

Reply:
I think this is the more elegant solution for a composable component like this one. as a user you just decide which header you want to display and share the data once, the rest of the component gets them through the context. With this we have less prop drilling and less interfaces. when we extend the possibilities of the header (for example i totally see the usecase for changing the displayed language (cURL -> node/kotlin/whatever). with this solution it is just one interface that has to change and you can use it right away. basically comes down to less propdrilling 🙂